zipline/extensions.py

```diff
@@ -0,0 +1,314 @@
+    interface : type
+        The base class for which to perform this operation
+    name : str
+        The name of the class to be retrieved.
+
+    Returns
+    -------
+    class : type
+        The desired class.
+    """
+    return get_registration_manager(interface).load(name)
+
+
+def class_registered(interface, name):
+    """
+    Whether or not the global dictionary of classes contains the
+    class with the specified name
+
+    Parameters
+    ----------
+    interface : type
+        The base class for which to perform this operation
+    name : str
+        The name of the class
+
+    Returns
+    -------
+    result : bool
+        Whether or not a given class is registered
+    """
+
+    return get_registration_manager(interface).class_registered(name)
+
+
+def register(interface, name, custom_class=None):
+    """
+    Registers a class for retrieval by the load method
+
+    Parameters
+    ----------
+    interface : type
+        The base class for which to perform this operation
+    name : str
+        The name of the subclass
+    custom_class : type
+        The class to register, which must be a subclass of the
+        abstract base class in self.dtype
+    """
+    if custom_class is None:
+        return partial(register, interface, name)
+
+    return get_registration_manager(interface).register(name, custom_class)
+
+
+def unregister(interface, name):
+    """
+    If a class is registered with the given name,
+    it is unregistered.
+
+    Parameters
+    ----------
+    interface : type
+        The base class for which to perform this operation
+    name : str
+        The name of the class to be unregistered.
+    """
+    get_registration_manager(interface).unregister(name)
+
+
+def clear(interface):
+    """
+    Unregisters all current registered classes
+
+    Parameters
+    ----------
+    interface : type
+        The base class for which to perform this operation
+    """
+    get_registration_manager(interface).clear()
+
+
+def get_registered_classes(interface):
+    """
+    A getter method for the dictionary of registered classes
+
+    Parameters
+    ----------
+    interface : type
+        The base class for which to perform this operation
+
+    Returns
+    -------
+    classes : dict
+        The dictionary of registered classes
+    """
+    return get_registration_manager(interface).get_registered_classes()
+
+
+def create_registration_manager(interface):
+    """
+
+    Parameters
+    ----------
+
+    Returns
+    -------
+    dtype : type
+        The data type specified/decorated
+    """
+    if interface in custom_types:
+        raise ValueError('there is already a RegistrationManager instance '
+                         'for the specified type')
+    custom_types[interface] = RegistrationManager(interface)
+    return interface
+
+
+extensible = create_registration_manager
+
+
+# A global dictionary for storing instances of RegistrationManager:
+custom_types = OrderedDict([])
```

Comment:
@llllllllll This was originally so that we would have an easy way of removing/modifying any of the RegistrationManager instances, but I don't think we would really have to do anything like that due to the fact that the user only has these options displayed when they start running zipline, right?